commit 8b9f0402b02a34d2e987936f134a20dc3effd34b
Author: Cosimo Cecchi <cosimoc@gnome.org>
Date:   Tue Sep 6 18:26:54 2011 -0400

    iconview: fix a typo while removing the a11y adjustment weak ref
    
    The code sets old_adj_ptr to the location containing the old weak ref,
    but then wants to remove a weak ref from &view->old_hadj, causing warnings
    when disposing the widget.

diff --git a/gtk/gtkiconview.c b/gtk/gtkiconview.c
index a252e0f..8b4a184 100644
--- a/gtk/gtkiconview.c
+++ b/gtk/gtkiconview.c
@@ -8089,7 +8089,7 @@ gtk_icon_view_accessible_set_adjustment (AtkObject      *accessible,
   if (*old_adj_ptr)
     {
       g_object_remove_weak_pointer (G_OBJECT (*old_adj_ptr),
-                                    (gpointer *)&view->old_hadj);
+                                    (gpointer *)old_adj_ptr);
       g_signal_handlers_disconnect_by_func (*old_adj_ptr,
                                             gtk_icon_view_accessible_adjustment_changed,
                                             accessible);
